google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -423,12 +441,37 @@ public Builder setMd5(String md5) {
       return this;
     }
 
+    public Builder setMd5FromHexString(String md5HexString) {
+      if (md5HexString == null) {
+        return this;
+      }
+      if (md5HexString.startsWith("0x")) {
+        md5HexString = md5HexString.replaceFirst("0x", "");
+      }
+      byte[] bytes = new BigInteger(md5HexString, 16).toByteArray();
+      this.md5 = "0x" + BaseEncoding.base64().omitPadding().encode(bytes);
```

Comment:
Why does this prepend `0x`? 

Could you add an integration that exercises the `setMd5/Crc32cFromHexString` methods?